tests/common/snappi_tests/common_helpers.py

```diff
@@ -1029,3 +1037,118 @@ def start_pfcwd_fwd(duthost, asic_value=None):
+        duthost.shell("sudo ip netns exec {} sonic-clear dropcounters \n".format(asic))
+        duthost.shell("sudo sonic-clear counters \n")
+        duthost.shell("sudo sonic-clear pfccounters \n")
+    else:
+        duthost.shell("sonic-clear queuecounters \n")
+        duthost.shell("sonic-clear dropcounters \n")
+        duthost.shell("sudo sonic-clear counters \n")
+        duthost.shell("sudo sonic-clear pfccounters \n")
+
+
+def interface_stats(duthost, port):
+    """
+    Get the Rx and Tx port utilization, throughput and pkts from SONiC CLI.
+    This is the equivalent of the "show interface counters" command.
+    Args:
+        duthost (Ansible host instance): device under test
+        port (str): port name
+    Returns:
+        i_stats (dict): key-value of various interface stats
+    """
+
+    i_stats = {}
+    dut_key = (duthost.hostname+'_'+port).lower()
+
+    n_raw_out = duthost.shell("portstat -ji {}".format(port))['stdout']
+    raw_out_stripped = re.sub(r'^.*?\n', '', n_raw_out, count=1)
+    raw_json = json.loads(raw_out_stripped)
+
+    # Drops, error and over are combined together as failures.
+    rx_err = ['RX_DRP', 'RX_ERR', 'RX_OVR']
+    tx_err = ['TX_DRP', 'TX_ERR', 'TX_OVR']
+    rx_fail = 0
+    tx_fail = 0
+    for m in rx_err:
+        rx_fail = rx_fail + int(raw_json[port].get(m).replace(',', ''))
+    for m in tx_err:
+        tx_fail = tx_fail + int(raw_json[port].get(m).replace(',', ''))
+
+    # Rx and Tx throughput.
+    thrput = raw_json[port].get('RX_BPS')
+    if thrput.split(' ')[1] == 'MB/s' and (thrput.split(' ')[0]) != '0.00':
+        i_stats[dut_key+'_rx_thrput_Mbps'] = float(thrput.split(' ')[0]) * 8
+    else:
+        i_stats[dut_key+'_rx_thrput_Mbps'] = 0
+    thrput = raw_json[port].get('TX_BPS')
+    if thrput.split(' ')[1] == 'MB/s' and (thrput.split(' ')[0]) != '0.00':
+        i_stats[dut_key+'_tx_thrput_Mbps'] = float(thrput.split(' ')[0]) * 8
+    else:
+        i_stats[dut_key+'_tx_thrput_Mbps'] = 0
+
+    # Rx and Tx packets.
+    i_stats[dut_key+'_rx_pkts'] = int(raw_json[port].get('RX_OK').replace(',', ''))
+    i_stats[dut_key+'_tx_pkts'] = int(raw_json[port].get('TX_OK').replace(',', ''))
+    i_stats[dut_key+'_rx_fail'] = rx_fail
+    i_stats[dut_key+'_tx_fail'] = tx_fail
+
+    return i_stats
+
+
+def get_queue_count_all_prio(duthost, port):
+    """
+    Get the egress queue count in packets and bytes for a given port and priority from SONiC CLI.
+    This is the equivalent of the "show queue counters" command.
+    Args:
+        duthost (Ansible host instance): device under test
+        port (str): port name
+    Returns:
+        queue_dict (dict): key-value with key=dut+port+prio and value=queue count
+    """
+    queue_dict = {}
+
+    # Preparing the dictionary for all 7 priority queues.
+    for priority in range(7):
+        dut_key = (duthost.hostname+'_'+port+'_prio_'+str(priority)).lower()
```

Comment:
same here. better to use multi level dictionary for easy access instead of customized key.

Reply:
Thanks for the comment. The flat dictionary was used because at the end of the test, entire dictionary was converted into data-frame. However, I have added the nested dictionaries for get_interface_stats, get_pfc_count and queue count function and then flattening it, for use in data-frame.